helix-core/src/main/java/org/apache/helix/model/CloudConfig.java

```diff
@@ -188,11 +184,17 @@ public Builder() {
     }
 
     /**
-     * Constructor with Cluster Name as input
-     * @param clusterName
+     * Instantiate with a pre-populated record
+     * @param record a ZNRecord corresponding to a cloud configuration
      */
-    public Builder(String clusterName) {
-      _clusterName = clusterName;
+    public Builder(ZNRecord record) {
+      _cloudEnabled = record.getBooleanField(CloudConfigProperty.CLOUD_ENABLED.name(), false);
+      _cloudProvider =
+          CloudProvider.valueOf(record.getSimpleField(CloudConfigProperty.CLOUD_PROVIDER.name()));
+      _cloudID = record.getSimpleField(CloudConfigProperty.CLOUD_ID.name());
+      _cloudInfoSources = record.getListField(CloudConfigProperty.CLOUD_INFO_SOURCE.name());
+      _cloudInfoProcessorName =
+          record.getSimpleField(CloudConfigProperty.CLOUD_INFO_PROCESSOR_NAME.name());
```

Comment:
We can keep a record as internal member.

Reply:
Done.